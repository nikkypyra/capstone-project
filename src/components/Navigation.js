import React from 'react'
import styled from 'styled-components/macro'
import { NavLink } from 'react-router-dom'
import { ReactComponent as Home } from '../icons/home.svg'
import { ReactComponent as Filter } from '../icons/filter.svg'

export default function Navigation() {
  return (
    <FooterStyled>
      <NavStyled exact to="/" activeClassName="chosen">
        <HomeStyled />
      </NavStyled>
      <NavStyled to="/filter" activeClassName="chosen">
        <FilterStyled />
      </NavStyled>
    </FooterStyled>
  )
}

const FooterStyled = styled.footer`
  position: fixed;
  bottom: 0;
  background: white;
  width: 100%;
  height: 48px;
  display: flex;
  justify-content: space-evenly;
`
const NavStyled = styled(NavLink)`
  display: flex;
  align-items: center;
  height: 48px;
  &.chosen {
    svg {
      fill: var(--primary);
    }
  }
`
const HomeStyled = styled(Home)`
  fill: var(--quaternary);
  height: 32px;
  width: 32px;
`
const FilterStyled = styled(Filter)`
  fill: var(--quaternary);
  height: 32px;
  width: 32px;
`
