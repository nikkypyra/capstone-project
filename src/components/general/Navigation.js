import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'
import { ReactComponent as Home } from '../../images/home.svg'
import { ReactComponent as Filter } from '../../images/filter.svg'
import { ReactComponent as Settings } from '../../images/settings.svg'

Navigation.propTypes = {
  onClick: PropTypes.func,
}

export default function Navigation({ onClick }) {
  return (
    <FooterStyled>
      <NavStyled
        to="/filter"
        activeClassName="selected"
        data-cy="filter"
        data-testid="filter"
      >
        <Filter onClick={onClick} />
      </NavStyled>
      <NavStyled exact to="/home" activeClassName="selected" data-cy="home">
        <Home onClick={onClick} />
      </NavStyled>
      <NavStyled
        to="/settings"
        activeClassName="selected"
        data-cy="settings"
        data-testid="settings"
      >
        <Settings onClick={onClick} />
      </NavStyled>
    </FooterStyled>
  )
}

const FooterStyled = styled.footer`
  position: fixed;
  bottom: 0;
  background: white;
  width: 100%;
  height: 48px;
  display: flex;
  justify-content: space-evenly;
`
const NavStyled = styled(NavLink)`
  display: flex;
  align-items: center;
  height: 48px;
  svg {
    fill: var(--quaternary);
    height: 32px;
    width: 32px;
  }
  &.selected {
    svg {
      fill: var(--primary);
    }
  }
`
