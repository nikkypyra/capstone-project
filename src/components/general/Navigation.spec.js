import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import Navigation from './Navigation'

afterEach(cleanup)

const onClick = jest.fn()

describe('<Navigation />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Navigation onClick={onClick} />
      </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
