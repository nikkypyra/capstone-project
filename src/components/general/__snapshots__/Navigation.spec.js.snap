// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render correctly 1`] = `
<DocumentFragment>
  <footer
    class="Navigation__FooterStyled-ug5o82-0 ggQCvf"
  >
    <a
      class="Navigation__NavStyled-ug5o82-1 euePKm"
      data-cy="home"
      href="/home"
    >
      <svg>
        home.svg
      </svg>
    </a>
    <a
      class="Navigation__NavStyled-ug5o82-1 euePKm"
      data-cy="filter"
      data-testid="filter"
      href="/filter"
    >
      <svg>
        filter.svg
      </svg>
    </a>
  </footer>
</DocumentFragment>
`;
